feat: woostack-init — workspace scaffold + scope-routed memory contract (increment A) (#154)
howarewoo/monorepo-template: skills/woostack-init/references/memory.md

5. **Stop.** Notes not matched in steps 3–4 are never loaded.

`recall.sh` — which orchestrates steps 2–4 — is the increment-B deliverable. It ships alongside its first consumer (the woostack-review migration) in increment B. **Increment A ships only the `scope-match.sh` primitive** (step 3's core) plus this documented procedure. Any consuming skill that wants to implement recall before increment B lands should follow this procedure manually, using `scope-match.sh` for step 3.

---

## 7. Scripts

The scripts live under `skills/woostack-init/scripts/` relative to the woostack repo root. In a consumer repo they are invoked via the path resolved by the agent when the woostack-init skill is available.

| Script | Usage |
|---|---|
| `scope-match.sh` | `printf '%s\n' <paths> \| bash scope-match.sh '<glob-spec>'` — prints matching paths from stdin; exits 0 if any matched, 1 if none. |
| `build-index.sh` | `bash build-index.sh [<memdir>]` — regenerates `<memdir>/MEMORY.md` from note frontmatter; defaults to `.woostack/memory`. |
| `doctor.sh` | `bash doctor.sh [<memdir>]` — lints the memory directory; warnings exit 0, errors exit 1. |

> All three scripts depend on `lib.sh` (frontmatter helpers `field()`, `note_body()`, `first_body_line()`) in the same directory. `doctor.sh` also calls `scope-match.sh`. Sourcing order: `lib.sh` first, then `scope-match.sh` if needed.

Author: **Script dependency claim wrong — scope-match has no lib.sh**

memory.md §7 states "All three scripts depend on `lib.sh`" but `scope-match.sh` does not source `lib.sh` at all — it is fully self-contained. Only `build-index.sh` and `doctor.sh` source `lib.sh`. A consumer or CI step treating `scope-match.sh` as dependent on `lib.sh` being co-located will be wrong, and the inaccuracy undermines trust in the contract for downstream users.

**Fix:** Change "All three scripts" to "`build-index.sh` and `doctor.sh`" so the sentence reads: "`build-index.sh` and `doctor.sh` depend on `lib.sh` (frontmatter helpers `field()`, `note_body()`, `first_body_line()`) in the same directory. `doctor.sh` also calls `scope-match.sh`. Sourcing order: `lib.sh` first, then `scope-match.sh` if needed."

```suggestion
`build-index.sh` and `doctor.sh` depend on `lib.sh` (frontmatter helpers `field()`, `note_body()`, `first_body_line()`) in the same directory. `doctor.sh` also calls `scope-match.sh`. Sourcing order: `lib.sh` first, then `scope-match.sh` if needed.
```

Author: Fixed in da21afd. Reworded §7: build-index.sh and doctor.sh source lib.sh; scope-match.sh is self-contained (sources nothing) and doctor.sh invokes it as a subprocess. Dropped the misleading 'sourcing order' line since scope-match is run as a subprocess, not sourced.
